lxd/apparmor/pyuefivars.go

```diff
@@ -0,0 +1,161 @@
+package apparmor
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/canonical/lxd/lxd/subprocess"
+	"github.com/canonical/lxd/lxd/sys"
+	"github.com/canonical/lxd/shared"
+)
+
+var pythonUefiVarsProfileTpl = template.Must(template.New("pythonUefiVarsProfile").Parse(`#include <tunables/global>
+profile "{{ .name }}" flags=(attach_disconnected,mediate_deleted) {
+  #include <abstractions/base>
+
+  # Python locations
+  /usr/bin/python* mixr,
+  /bin**/*.py r,
+  /usr/lib/python** r,
+  /lib/python** r,
+  /**/pyvenv.cfg r,
+
+{{- if .snap }}
+  # Snap-specific libraries
+  /snap/lxd/*/lib/**.so* mr,
+
+  # Snap-specific Python locations
+  /snap/lxd/*/bin**/*.py r,
+  /snap/lxd/*/usr/lib/python** r,
+  /snap/lxd/*/lib/python** r,
+{{- end }}
+
+{{if .libraryPath -}}
+  # Entries from LD_LIBRARY_PATH
+{{range $index, $element := .libraryPath}}
+  {{$element}}/** mr,
+{{- end }}
+{{- end }}
+}
+`))
+
+type nopWriteCloser struct {
+	io.Writer
+}
+
+func (*nopWriteCloser) Close() error {
+	return nil
+}
+
+// PythonUefiVars runs pyuefivars with an AppArmor profile based on the efiVarsPath supplied.
+func PythonUefiVars(sysOS *sys.OS, stdin io.Reader, stdout io.Writer, efiVarsPath string) error {
```

Comment:
In general Go suggests using upper case for acronyms in symbols, so here would normally be called PythonUEFIVars()

Reply:
Fixed, thanks!